src/transform/src/analysis/equivalences.rs

```diff
@@ -714,4 +718,122 @@ impl EquivalenceClasses {
         }
         false
     }
+
+    /// Returns a map that can be used to replace (sub-)expressions.
+    pub fn reducer(&self) -> BTreeMap<&MirScalarExpr, &MirScalarExpr> {
+        self.classes
+            .iter()
+            .flat_map(|c| c.iter().map(move |e| (e, &c[0])))
+            .collect()
+    }
+}
+
+trait ExpressionReducer {
+    fn replace(&self, expr: &mut MirScalarExpr) -> bool;
+    fn reduce_expr(&self, expr: &mut MirScalarExpr) -> bool {
+        let mut simplified = false;
+        simplified = simplified || self.reduce_child(expr);
+        simplified = simplified || self.replace(expr);
+        simplified
+    }
+    fn reduce_child(&self, expr: &mut MirScalarExpr) -> bool {
```

Comment:
This seems to be a duplicate code fragment: there is an other, identical `reduce_child` in the same file. It's not on the trait, though, so `self` is different, so maybe this is intentional? I'm not sure.

Reply:
Right, so it is different, and it is intentional. This method, and the others in the trait, are meant to sub in for those methods on Equivalences. However, we haven't pivoted the whole codebase over from Equivalences::replace to ExpressionReducer::replace. Internally, minimize_once uses the ExpressionReducer version, but externally (e.g. in EquivalencePropagation the interface hasn't changed yet (the first goal is to improve minimize rather than uses of Equivalences).